staging/src/k8s.io/legacy-cloud-providers/install/providers.go

```diff
@@ -16,7 +16,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cloudprovider
+package install
```

Comment:
I understand the need to move this to staging. However that essentially makes it public and we don't want to support it. Can we add a deprecation warning and a statement that binaries Kubernetes core binaries such as Kubelet should not be linking in cloud providers and that non core binaries should explicitly specify which cloud provider they are linking with.

Reply:
I'm having a hard time seeing why we want a "link all providers" package in k8s.io/legacy-cloud-providers, when we don't actually want anything outside k8s.io/kubernetes and k8s.io/cloud-controller-manager doing that, right?

Reply:
I agree and had the same comment here https://github.com/kubernetes/kubernetes/pull/91218#discussion_r427516521 

For the cloud-controller-manager case where we can't import `k8s.io/kubernetes` we can import the legacy providers individually. The other components can continue importing `k8s.io/kubernetes/pkg/cloudprovider/providers`.

Reply:
Would it be better to go with this way: https://github.com/kubernetes/kubernetes/pull/92880?